src/components/Extensions/Details/DetailCard/Tabs/EventsConfig.tsx

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+
+import { Typography } from '@rmwc/typography';
+
+import { Accordion } from '../../../../common/Accordion';
+import { Markdown } from '../../../../common/Markdown';
+import { useExtension } from '../../../api/useExtension';
+import { ParamType } from '../../../models';
+import styles from './EventsConfig.module.scss';
+import { ParamValue } from './ParamValue';
+
+export function EventsConfig() {
+  const extension = useExtension()!;
+  const eventsByPrefix: Map<string, string[]> = new Map(); // maps prefix ==> list of types
+  const eventDescriptionMapping: Map<string, string> = new Map(); // maps type ==> description
+  let allowedEventTypesDiv;
+  let channelLocationDiv;
+  !!extension.events &&
+    extension.events.forEach((event) => {
+      eventDescriptionMapping.set(event.type, event.description);
+    });
+  if (!!extension.allowedEventTypes && extension.allowedEventTypes.length > 0) {
+    extension.allowedEventTypes.forEach((eventType) => {
+      const prefix = eventType.split('.').slice(0, -1).join('.');
+      if (!eventsByPrefix.get(prefix)) {
+        eventsByPrefix.set(prefix, [eventType]);
+      } else {
+        eventsByPrefix.get(prefix)?.push(eventType);
+      }
+    });
+    const allowedEventTypeElements = Array.from(eventsByPrefix.keys())
+      .sort()
+      .map((prefix) => {
+        const events = eventsByPrefix.get(prefix)?.map((event) => {
+          return (
+            <div key={`${event}`} className={styles.eventValue}>
+              <ParamValue value={event} type={ParamType.STRING} />
+              <Typography use="body1" theme="secondary">
+                {eventDescriptionMapping.get(event) || 'Not Specified'}
+              </Typography>
+            </div>
+          );
+        });
+        return (
+          <div key={`${prefix}`} className={styles.eventPrefixValue}>
+            <ParamValue value={prefix} type={ParamType.STRING} />
+            {events}
+          </div>
+        );
+      });
+    allowedEventTypesDiv = (
+      <div key="allowedEventTypes" className={styles.paramWrapper}>
+        <Accordion
+          expansionLabel="Description"
+          title={
+            <Typography use="body2" theme="secondary">
+              Types of events
+            </Typography>
+          }
+        >
+          <Typography
+            className={styles.description}
+            use="caption"
+            tag="div"
+            theme="secondary"
+          >
+            <Markdown>
+              Which events do you want to be emitted from this extension? Only
+              selected events will be published to Eventarc, all other events
+              will be ignored and will not be counted against Eventarc usage.
+            </Markdown>
+          </Typography>
+        </Accordion>
+        {allowedEventTypeElements}
+      </div>
+    );
+  }
+  if (!!extension.eventarcChannel) {
```

Comment:
Making channelLocationDiv a const w/ a ternary on eventarcChannel would allow us to use a const :)